modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -257,8 +257,13 @@ def groupby_reduce(
             )
         else:
             mapped_partitions = cls.map_partitions(partitions, map_func)
+        num_splits = min(len(partitions), NPartitions.get())
```

Comment:
let's just add an in-code comment explaining this logic and the PR is good to go from my side